### ISSUE TITLE
Decouple model traits from measure traits?

### ISSUE THREAD

**Issue by ablaom** [alan-turing-institute/MLJBase.jl#495]
Presently there are some measure traits which are overloaded model traits. Specifically these are:
`name`, `target_scitype`, `supports_weights`, `prediction_type`, and `docstring`. To this list, [this PR](https://github.com/alan-turing-institute/MLJBase.jl/pull/464) would require us to add `supports_sample_weights` (currently a measure trait but needs to be a model trait as well).

**edit** Another method that is used by both models and traits is `info`

The problem with this state of affairs is that model traits are defined in MLJModelInterface, but we plan to have StatisticalMeasures as a separate packages (see #416). Although MLJModelInterface is very lightweight, to have it as a dependency of StatisticalMeausures.jl is like to jeopardise the latter's adoption as a general use package, independent of the MLJ stack. 

Here are two possible resolutions:

1.  Introduce a tiny ultra-lightweight package `MachineLearningMethodStubs.jl` (or whatever) to contain only stubs for traits used throughout the MLJ stack. It becomes a common dependency of MLJModelInterface and the eventual StatisticalMeasures.jl package. 

2. We move the current measures code into a module (or make it a package already) and keep its traits private. That is, their use will need to be qualified in, say, MLJBase and elsewhere.

Probably option 1 is the least disruptive; we could do this and still postpone the creation of `StatisticalMeasures`, which I expect to be a little painful. 

Thoughts or other ideas anyone?

cc @DilumAluthge @ven-k @tlienart

**Comment by ven-k** [comment]
How about, we define functions privately and overload only when it is already available for non-measure functions and keep this issue open to track these traits that need to be cleaned up? (; as we will have a lot of measure-code moved this might not be a hassle). I will close[ this](https://github.com/alan-turing-institute/MLJModelInterface.jl/pull/75). And once the brainstorming for `StatisticalMeasures` is done we go ahead with **2** in your comment. 
This might be messy for a while though.

However if there are more use-cases for `MLMethodsStub` (like ease for MLJ development when StatisticalMeasures are involved), it sounds good.

**Comment by ablaom** [comment]
Okay, after some discussion on slack and at [this thread](https://github.com/JuliaStats/StatsBase.jl/issues/625#issuecomment-759287550), I propose we create our own StatisticalTraits.jl package to contain fallback definitions of model traits and measures traits. 

To do:

- [ ] Create and register new StatisticalTraits.jl package with default trait definitions, and a stub for `info`. 

- [ ] Update MLJModelInterface to import traits from above

- [ ] Update MLJScientificTypes to import `info` from above

- [ ] Check if MLJBase needs any changes

I will roll out the changes only after testing all repos locally. 

Self-assigning.
